app/rpc/namespaces/eth/tx_pool.go

```diff
@@ -0,0 +1,306 @@
+		api.logger.Error("eth_sendRawTransaction txPool err:", err.Error())
+		api.txPool.mu.Unlock()
+		return common.Hash{}, err
+	}
+	api.txPool.mu.Unlock()
+
+	return common.HexToHash(strings.ToUpper(hex.EncodeToString(tmhash.Sum(txBytes)))), nil
+}
+
+func (pool *TxPool) CacheAndBroadcastTx(api *PublicEthereumAPI, address common.Address, tx *evmtypes.MsgEthereumTx) error {
+	// get currentNonce
+	pCurrentNonce, err := api.GetTransactionCount(address, rpctypes.PendingBlockNumber)
+	if err != nil {
+		return err
+	}
+	currentNonce := uint64(*pCurrentNonce)
+
+	if tx.Data.AccountNonce < currentNonce {
+		return fmt.Errorf("AccountNonce of tx is less than currentNonce in memPool: AccountNonce[%d], currentNonce[%d]", tx.Data.AccountNonce, currentNonce)
+	}
+
+	if tx.Data.AccountNonce > currentNonce+viper.GetUint64(TxPoolSliceMaxLen) {
+		return fmt.Errorf("AccountNonce of tx is bigger than txPool capacity, please try later: AccountNonce[%d]", tx.Data.AccountNonce)
+	}
+
+	if err = pool.insertTx(address, tx); err != nil {
+		return err
+	}
+
+	// update DB
+	if err = pool.writeTxInDB(address, tx); err != nil {
+		return err
+	}
+
+	return pool.continueBroadcast(api, currentNonce, address)
+}
+
+func (pool *TxPool) update(index int, address common.Address, tx *evmtypes.MsgEthereumTx) error {
+	txsLen := len(pool.addressTxsPool[address])
+	if index >= txsLen {
+		pool.addressTxsPool[address] = append(pool.addressTxsPool[address], tx)
+	} else {
+		tmpTx := make([]*evmtypes.MsgEthereumTx, len(pool.addressTxsPool[address][index:]))
+		copy(tmpTx, pool.addressTxsPool[address][index:])
+
+		pool.addressTxsPool[address] =
+			append(append(pool.addressTxsPool[address][:index], tx), tmpTx...)
+	}
+	return nil
+}
+
+// insert the tx into the txPool
+func (pool *TxPool) insertTx(address common.Address, tx *evmtypes.MsgEthereumTx) error {
+	// if this is the first time to insertTx, make the cap of txPool be TxPoolSliceMaxLen
+	if _, ok := pool.addressTxsPool[address]; !ok {
+		pool.addressTxsPool[address] = make([]*evmtypes.MsgEthereumTx, 0, viper.GetUint64(TxPoolSliceMaxLen))
+	}
+	index := 0
+	for index < len(pool.addressTxsPool[address]) {
+		// the tx nonce has in txPool, drop duplicate tx
+		if tx.Data.AccountNonce == pool.addressTxsPool[address][index].Data.AccountNonce {
+			return fmt.Errorf("duplicate tx, this AccountNonce of tx has been in txPool. AccountNonce[%d]", tx.Data.AccountNonce)
+		}
+		// find the index to insert
+		if tx.Data.AccountNonce < pool.addressTxsPool[address][index].Data.AccountNonce {
+			break
+		}
+		index++
+	}
+	// update txPool
+	return pool.update(index, address, tx)
+}
+
+// iterate through the txPool map, check if need to continue broadcast tx and do it
+func (pool *TxPool) continueBroadcast(api *PublicEthereumAPI, currentNonce uint64, address common.Address) error {
+	i := 0
+	txsLen := len(pool.addressTxsPool[address])
+	var err error
+	for ; i < txsLen; i++ {
```

Comment:
It is better to add a nonce security check. The nonce of the front tx is required to be greater than or equal to currentNonce, otherwise the clear operation